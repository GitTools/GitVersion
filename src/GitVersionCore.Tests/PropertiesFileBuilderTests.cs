using System;
using ApprovalTests;
using GitVersion;
using NUnit.Framework;

[TestFixture]
public class PropertiesFileBuilderTests
{
    [Test]
    public void Props()
    {
        var semanticVersion = new SemanticVersion
            {
                Major = 1,
                Minor = 2,
                Patch = 3,
                PreReleaseTag = "unstable4",
                BuildMetaData = new SemanticVersionBuildMetaData(5, "feature1", "commitSha",DateTimeOffset.Parse("2014-03-06 23:59:59Z"))
            };
        var variables = VariableProvider.GetVariablesFor(semanticVersion, AssemblyVersioningScheme.MajorMinorPatch, VersioningMode.ContinuousDelivery, "ci", false);
        var props = PropertiesOutputFormatter.ToProperties(variables);
        Approvals.Verify(props);
    }
}
