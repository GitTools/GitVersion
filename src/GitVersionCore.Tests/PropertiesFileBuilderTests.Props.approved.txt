gitversion.Major=1
gitversion.Minor=2
gitversion.Patch=3
gitversion.PreReleaseTag=unstable.4
gitversion.PreReleaseTagWithDash=-unstable.4
gitversion.BuildMetaData=5
gitversion.FullBuildMetaData=5.Branch.feature1.Sha.commitSha
gitversion.MajorMinorPatch=1.2.3
gitversion.SemVer=1.2.3-unstable.4
gitversion.LegacySemVer=1.2.3-unstable4
gitversion.LegacySemVerPadded=1.2.3-unstable0004
gitversion.AssemblySemVer=1.2.3.0
gitversion.FullSemVer=1.2.3-unstable.4+5
gitversion.InformationalVersion=1.2.3-unstable.4+5.Branch.feature1.Sha.commitSha
gitversion.BranchName=feature1
gitversion.Sha=commitSha
gitversion.NuGetVersionV2=1.2.3-unstable0004
gitversion.NuGetVersion=1.2.3-unstable0004
gitversion.CommitDate=2014-03-06
